api/v1alpha3/condition_consts.go

```diff
@@ -69,6 +69,18 @@ const (
 	WaitingForDataSecretFallbackReason = "WaitingForDataSecret"
 )
 
+// Conditions and condition Reasons for the MachineSet object
+
+const (
+	// ReplicasUpdated reports the last time any replica count was updated
+	// in the MachineSet status.
+	// Only LastTransitionTime for this condition changes, all other fields
```

Comment:
wondering if this might confuse the users, because for all the other conditions LastTransitionTime is
the Last time the condition transitioned from one status to another, while in this case we are tracking when the replica number changed (which is not a transition from one status to another)

Reply:
I think it will be more confusing without the reason and message.  I think it's pretty obvious what it's doing in present form.

Reply:
This is so far the only CAPI condition that requires reason and message to document the true state, so probably it so obvious to someone else.
Nevertheless, no matter of reason and message, it is still open the point that in this case we are altering the meaning of the LastTransitionTime with respect to all the other conditions.